src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseParticipationService.java

```diff
@@ -147,13 +144,14 @@ public ProgrammingExerciseParticipation findProgrammingExerciseParticipationWith
      * - Template/SolutionParticipations should only be accessible for users with at least the role TA in the courses.
      *
      * @param participation to check permissions for.
+     * @param user          the current user.
      * @return true if the user can access the participation, false if not. Also returns false if the participation is not from a programming exercise.
      */
-    public boolean canAccessParticipation(@NotNull ProgrammingExerciseParticipation participation) {
+    public boolean canAccessParticipation(@NotNull ProgrammingExerciseParticipation participation, User user) {
```

Comment:
Note: related changes in #6384 for this method and callers, merge conflicts will arise for one of the PRs. The actual refactorings concern different authorization checks, though.

Reply:
Thanks for the heads up! I need to have the user as an extra parameter here, because the method will be called by the local VC system where I retrieve the user from the Authorization header of a Git request and `userRepository.getUserWithGroupsAndAuthorities()` does not work.